### PR TITLE
Fix transformation sting

### LINKED ISSUE #9112: changeling transformatian sting not working
Issue reported from  Round ID: 44249 (BeeStation Sage - Roleplay)

Reporting client version: 514.1589

able to sting people but doesnt make them transform
was told to report this here by admin

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

I'm a dummy and didn't realize `has_status_effect` would return `FALSE` instead of `null` whenever there's no status effect...

Fixes https://github.com/BeeStation/BeeStation-Hornet/issues/9112

## Why It's Good For The Game

it's supposed to work

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
fix: Changeling transformation sting works again.
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
Supposedly closes #9112

File: code/modules/antagonists/changeling/powers/tiny_prick.dm [hunk 1]
```diff
@@ -114,7 +114,7 @@
 		if(ismonkey(C))
 			C = C.humanize(TR_KEEPITEMS | TR_KEEPIMPLANTS | TR_KEEPORGANS | TR_KEEPDAMAGE | TR_KEEPVIRUS | TR_DEFAULTMSG | TR_KEEPAI)
 		var/datum/status_effect/ling_transformation/previous_transformation = C.has_status_effect(STATUS_EFFECT_LING_TRANSFORMATION)
-		C.apply_status_effect(STATUS_EFFECT_LING_TRANSFORMATION, new_dna, previous_transformation?.original_dna)
+		C.apply_status_effect(STATUS_EFFECT_LING_TRANSFORMATION, new_dna, (previous_transformation && istype(previous_transformation)) ? previous_transformation.original_dna : null)
```
Comment (Archanial) [inline code/modules/antagonists/changeling/powers/tiny_prick.dm]: If I am reading the code right previous_transformation will never be null and is either the type or false. You don't need to  check for null here.

**Comment by itsmeow** [comment]
Does this also fix hallucination sting not working?

**Comment by itsmeow** [comment]
No it does not. sad

**Review by itsmeow — APPROVED** [review]

**Pull request merged**